### PR TITLE
IACO-552: Document Purchase - sample backend

### PR DESCRIPTION
**Pull request by grozen**
Added some documentation to the sample backend to make things clearer. See the docs rendered [here](https://github.com/klarna/sample-app-backend/tree/feature/IACO-552-document-purchase).
Also ultra-commented the code for non Rubyists.

### CODE REVIEW

File: backend.rb [hunk 1]
```diff
@@ -25,13 +26,16 @@ class Backend < Sinatra::Base
     }
   }
 
+  # Handle POST requests to '/pay'
   post '/pay' do
+    # Make key-value pairs from the JSON body available in the 'params' hash
```
Comment (tomerlevi444) [inline backend.rb:35]: `Make key-value pairs from the JSON body available in the 'params' hash`  ~> `Insert the key-value pairs from the JSON body into the 'params' hash` ?

File: backend.rb [hunk 2]
```diff
@@ -51,12 +55,16 @@ class Backend < Sinatra::Base
     authorize_payment_response = HTTParty.post(authorize_url, authorize_request_options)
 
     if (authorize_payment_response && authorize_payment_response.code == 201)
+      # The purchase was authorized (and has essentially been created as a
+      # resource available at the location specified in the authorization
+      # requests's response). Capture it.
```
Comment (tomerlevi444) [inline backend.rb:75]: There needs to be a separate explanation about the meaning of authorizing an order and capturing it.

**Comment by tomerlevi444** [comment]
Approved!

**Comment by yuval-netanel** [comment]
Approved! 
Is there a way to print the request to the log?

File: backend.rb (hunk 1 above)
Comment (grozen) [inline backend.rb:35]: Isn't that pretty much the same thing? Not that I mind, really.

File: backend.rb (hunk 2 above)
Comment (grozen) [inline backend.rb:75]: That's something that should be available at developers.klarna.com at some point. It's the APIs documentation, and therefor not the best fit for anything we have on github.

**Comment by grozen** [comment]
@yuval-netanel that's actually a good idea. I'll add logging of incoming/outgoing requests to the console.

File: backend.rb (hunk 1 above)
Comment (tomerlevi444) [inline backend.rb:35]: It's more intuitive for a non-rubyist

File: backend.rb [hunk 3]
```diff
@@ -12,11 +13,15 @@ class Backend < Sinatra::Base
     register Sinatra::Reloader
   end
 
+  requestLog = Logger.new STDOUT
```
Comment (rauchy) [inline backend.rb:16]: camel_case?

Comment (grozen) [inline backend.rb:16]: That's what you get when you mix your alcohol.

**Pull request merged**